sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1325,25 +1325,54 @@ class Analyzer(
      *
      * Note : In this routine, the unresolved attributes are resolved from the input plan's
      * children attributes.
+     *
+     * @param e The expression need to be resolved.
+     * @param q The LogicalPlan whose children are used to resolve expression's attribute.
+     * @param trimAlias When true, we will trim Struct field alias. When isTopLevel = true,
```

Comment:
there is no `isTopLevel` parameter here.
```
When true, trim unnecessary alias of `GetStructField`. Note that, we cannot trim
the alias of top-level `GetStructField`, as we should resolve `UnresolvedAttribute` to
a named expression. The caller side can trim the alias of top-level `GetStructField` if
it's safe to do so.
```

Just let the caller side know that this method only trims alias of non-top-level `GetStructField`, and why.

Reply:
For `innerResolve`, the `isTopLevel` parameter is just an internal state to indicate top-level `GetStructField`, too trivial to comment.

Reply:
> For `innerResolve`, the `isTopLevel` parameter is just an internal state to indicate top-level `GetStructField`, too trivial to comment.

Yea, above comment is enough